pkg/onepassword/client/client.go

```diff
@@ -0,0 +1,55 @@
+package client
+
+import (
+	"errors"
+	"os"
+
+	"github.com/go-logr/logr"
+
+	"github.com/1Password/onepassword-operator/pkg/onepassword/client/connect"
+	"github.com/1Password/onepassword-operator/pkg/onepassword/client/sdk"
+	"github.com/1Password/onepassword-operator/pkg/onepassword/model"
+)
+
+// Client is an interface for interacting with 1Password items and vaults.
+type Client interface {
+	GetItemByID(vaultID, itemID string) (*model.Item, error)
+	GetItemsByTitle(vaultID, itemTitle string) ([]model.Item, error)
+	GetFileContent(vaultID, itemID, fileID string) ([]byte, error)
+	GetVaultsByTitle(title string) ([]model.Vault, error)
+}
+
+type Config struct {
+	Logger  logr.Logger
+	Version string
+}
+
+// NewFromEnvironment creates a new 1Password client based on the provided configuration.
+func NewFromEnvironment(cfg Config) (Client, error) {
+	connectHost, _ := os.LookupEnv("OP_CONNECT_HOST")
+	connectToken, _ := os.LookupEnv("OP_CONNECT_TOKEN")
+	serviceAccountToken, _ := os.LookupEnv("OP_SERVICE_ACCOUNT_TOKEN")
+
+	if connectHost != "" && connectToken != "" && serviceAccountToken != "" {
```

Comment:
`[nit]`

Shouldn't we fail if at least one Connect related environment is set alongside the service account one? Having some weird configuration working (e.g. Connect host + service account token) can be confusing and give the wrong impression.

Reply:
We already throw an error in this case. 

Connect configuration is fully setup only if both HOST and TOKEN are provided.

If at least one Connect env var set, it will try to use Service Account (if it set) or error if Service Account is empty.